app/controllers/api/v1/accounts/articles_controller.rb

```diff
@@ -0,0 +1,38 @@
+class Api::V1::Accounts::ArticlesController < Api::V1::Accounts::BaseController
+  before_action :portal
+  before_action :fetch_article, except: [:index, :create]
+
+  def index
+    @articles = @portal.articles.where(status: :published)
+    @articles.search(params[:payload]) if params[:payload].present?
```

Comment:
What is the payload here? Can we use permitted_params? 

Please add a spec for the same.

Reply:
For index? We can use permitted params, but we do it by searching based on params name, so there won't be any injection while searching.

Reply:
Basically this would be the payload for search articles 

```
{
    "payload": { 
        "locale": "es",
        "category_slug": "agent-category-1"
    }
}
```